### PR TITLE
Bump our deps in the crypto crate

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/identities/device.rs [hunk 1]
```diff
@@ -1028,4 +1029,53 @@ pub(crate) mod tests {
...
+                            }
+                        },
+                        "unsigned":{
+                            "device_display_name":"Alice's mobile phone"
+                        }
+                },
+                "deleted":false,
+                "trust_state":"Verified",
+                "withheld_code_sent":false,
+                "first_time_seen_ts":1696931068314u64
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/identities/device.rs:1059]: Lots of missing spaces after `:` field / value separator, would be nice to have consistent formatting.

Comment (poljar) [reply crates/matrix-sdk-crypto/src/identities/device.rs]: Right: https://github.com/matrix-org/matrix-rust-sdk/pull/2688/commits/7bebab85eda32b6c41aeea4fc1ef9ff898ec01af.

File: crates/matrix-sdk-crypto/src/identities/device.rs [hunk 2]
```diff
@@ -1028,4 +1029,53 @@ pub(crate) mod tests {
...
     }
+
+    #[test]
+    fn deserialize_device() {
+        let user_id = user_id!("@example:localhost");
+        let device_id = "BNYQQWUMXO";
+
+        let device = json!({
+            "inner":
+                {
```
[the remark this answers is not in the document]
Comment (poljar) [reply crates/matrix-sdk-crypto/src/identities/device.rs]: Fixed in https://github.com/matrix-org/matrix-rust-sdk/pull/2688/commits/7bebab85eda32b6c41aeea4fc1ef9ff898ec01af.

**Comment by poljar** [comment]
> We also depend on the older version of pbkdf in matrix-sdk-store-encryption, matrix-sdk-crypto-ffi still.

Doh, forgot about those two, thanks for reminding me: https://github.com/matrix-org/matrix-rust-sdk/pull/2688/commits/49273d63f53f8d73bdae6f37c7fa4ebd6ae1ec07.

**Review by jplatte — APPROVED** [review]
💚

**Pull request merged**